sinon/lib/spy.py

```diff
@@ -187,36 +179,43 @@ def lastCall(self): #pylint: disable=invalid-name
         lastIndex = len(self.__get_wrapper().call_list) - 1
         return self.getCall(lastIndex)
 
-    def calledBefore(self, obj): #pylint: disable=invalid-name
-        """
-        Comparing two spys' functions order in CALLQUEUE
-        Eg.
-            a()
-            b()
-            spy_a.calledBefore(spy_b) will return True, because a is called before b
-            a()
-            spy_b.calledBefore(spy_a) will return True, because a is called after b
-        Return: Boolean
+    def calledBefore(self, spy): #pylint: disable=invalid-name
         """
-        idx = self.get_callqueue_idx()
-        idx2 = obj.get_callqueue_idx()
+        Compares the order in which two spies were called
 
-        if len(idx) == 0:
-            ErrorHandler.callqueue_is_empty_error()
-        if Wrapper.CALLQUEUE:
-            return True if min(idx) < max(idx2) else False
+        E.g.
+            spy_a()
+            spy_b()
+            spy_a.calledBefore(spy_b) # True
+            spy_b.calledBefore(spy_a) # False
+            spy_a()
+            spy_b.calledBefore(spy_a) # True
 
-    def calledAfter(self, obj): #pylint: disable=invalid-name
+        Args: a Spy to compare with
+        Return: Boolean True if this spy's first call was called before the given spy's last call
         """
-        Same as calledBefore
+        thisCall = self.firstCall or False
+        givenCall = spy.lastCall or False
+        return (thisCall and not givenCall) or (thisCall and givenCall and thisCall.callId < givenCall.callId)
```

Comment:
Note: https://stackoverflow.com/questions/3914667/false-or-none-vs-none-or-false
```python
>>> print(None or False)
False
>>> print(False or None)
None
```

Considering the maintainability, I am not sure whether this coding style is a good idea. 
I suggest you to put some comment to describe you logic.

```
return ((thisCall and not givenCall) #if givenCall is not called
    or (thisCall and givenCall and thisCall.callId < givenCall.callId)) #if thisCall and givenCall are called, then compare callId
```